tests/test_core.py

```diff
@@ -8504,6 +8506,10 @@ def make_run(name, emcc_args, settings=None, env=None):
     env = {}
   if settings is None:
     settings = {}
+  if settings:
+    # Until we create a way to specify link-time settings separately from compile-time settings
+    # we need to pass this flag here to avoid warnings from compile-only commands.
+    emcc_args.append('-Wno-unused-command-line-argument')
```

Comment:
Sorry for commenting an old PR but can you elaborate on what this means?

Context: I found '-Wno-unused-command-line-argument' is passed for lsan and asan but not ubsan, because, unlike asan or lsan, ubsan does not pass any `settings` here,
https://github.com/emscripten-core/emscripten/blame/a9b347bfcabda59a5edff60ee18b8a0ab70aa9dc/test/test_core.py#L9782-L9786
and `-fsanitize=` is ignored when compiling `.s` file, as in 
https://github.com/emscripten-core/emscripten/blame/a9b347bfcabda59a5edff60ee18b8a0ab70aa9dc/test/test_core.py#L9566

As a result `ubsan.test_externref` fails. Our test suites CI (https://ci.chromium.org/ui/p/emscripten-releases/builders/luci.emscripten-releases.ci/linux-test-suites) skips this test so it wasn't detected. This is not related to the update of compiler-rt to LLVM 18; this is already failing.

Anyway, why should we pass any `settings`, whatever that is, to get this `-Wno-unused-command-line-argument`?

Reply:
Within the test framework we didn't have a great way to specify that a certain setting is a link time vs compile time setting.

Actually this recently improve a little I think `get_emcc_args` no takes `compile_only` flag which can be used to get just the compile flags and not the link flags.

Is `-fsanitize=undefined` a link-only flag?  I thought it was used during compilation so I'm struggling to see why we would be seeing ` clang: error: argument unused during compilation: '-fsanitize=undefined' [-Werror,-Wunused-command-line-argument]`

Reply:
Oh I see.. its because its an assembly file.

Reply:
We should remove `self.get_emcc_args` from the compile command for that `.s` file?

Reply:
If we remove `self.get_emcc_args` from this test, wouldn't `asan.test_externref` and `lsan.test_externref` also fail, because they wouldn't be able to get `-Wno-unused-command-line-argument` either? (and `ubsan.test_externref` will stay failing)

I think what we should do is to pass `-Wno-unused-command-line-argument` to ubsan too.

Reply:
But its only needed for this one test because it happens to be written in assembly.   Perhaps you can add `-Wno-unused-command-line-argument` to this one test?

Reply:
.s files cannot be sanitized anyway, so how about just adding `no_sanitize`? #21715

Reply:
Yup that works too